lib/redis_options.rb

```diff
@@ -16,8 +16,8 @@ def url
 
   memoize \
   def url_without_db
-    case ENV.fetch('RAILS_ENV', nil)
-    when 'development', 'test', nil then 'redis://localhost:6379'
```

Comment:
I think that all of the other changes in this PR are pure refactors (code changes not expected to change behavior). I think that this is the one change that actually changes behavior (or, at least, would change behavior, if RAILS_ENV is ever `nil`, but I think it never should be, which is why I'm making this change).